src/cli/peakdetector/peakdetectorcli.cpp

```diff
@@ -1067,6 +1067,12 @@ QString PeakDetectorCLI::uploadToPolly(QString jsPath,
     // set jspath and nodepath for _pollyIntegration library .
     _pollyIntegration->jsPath = jsPath;
     _pollyIntegration->nodePath = nodePath;
+    if (!_pollyIntegration->checkNodeExecutable()) {
+        _log->error() << "NodeJS was not found on this system. "
+                      "Please install NodeJS and try again."
+                      << std::flush;
```

Comment:
If the user installs NodeJS on their system using some package manager, they very likely will still not be able to use the CLI. Although this does hint to what the actual problem is, so we can leave this as is.

Reply:
Which was the case with me too. Although the node_bin thing is a special case for Mac and I could handle that here. Let me see.